### PR TITLE
OS-8174 lay groundwork for GCC9

### PR DESCRIPTION
**Pull request by jlevon**

### CODE REVIEW

File: Makefile.defs [hunk 1]
```diff
@@ -29,35 +29,49 @@ FETCH_BASE =	https://us-east.manta.joyent.com/Joyent_Dev/public/illumos-extra
...
 # 'configure' in smartos-live.
-GCC4.host =	$(HOST_GCC4_PREFIX)/gcc
-GXX4.host =	$(HOST_GCC4_PREFIX)/g++
+GCC4.host =	$(HOST_GCC4_PREFIX)/gcc -m32
+GXX4.host =	$(HOST_GCC4_PREFIX)/g++ -m32
+
+PATH =		$(STRAPPROTO)/usr/bin:/usr/bin:/usr/sbin:/sbin:/opt/local/bin
+
+MAKE =		gmake
+MAX_JOBS =	128
```
Comment (timfoster) [inline Makefile.defs:72]: I wonder whether this should be ?= to allow users to pass in their own MAX_JOBS value? https://github.com/joyent/smartos-live#controlling-maximum-number-of-jobs

Comment (jlevon) [reply Makefile.defs:72]: Maybe, but I'd rather not try to fix this as part of this change (this line is just moved, not added)

File: Makefile.defs [hunk 2]
```diff
@@ -29,35 +29,49 @@ FETCH_BASE =	https://us-east.manta.joyent.com/Joyent_Dev/public/illumos-extra
...
+GCC.host.32 =	$(GCC.host_prefix)/gcc -m32
+GXX.host.32 =	$(GCC.host_prefix)/g++ -m32
+GCC.host.64 =	$(GCC.host_prefix)/gcc -m64
+GXX.host.64 =	$(GCC.host_prefix)/g++ -m64
 
 # We need gcc 4.x in order to build our copy of gcc 4.4. This is set by
 # 'configure' in smartos-live.
-GCC4.host =	$(HOST_GCC4_PREFIX)/gcc
-GXX4.host =	$(HOST_GCC4_PREFIX)/g++
+GCC4.host =	$(HOST_GCC4_PREFIX)/gcc -m32
```
Comment (timfoster) [inline Makefile.defs:66]: Should we comment here that this _must_ be -m32, and if it changes, be sure to also change ./gcc4/Makefile which does `GCC.host.32 = $(GCC4.host)`

File: Makefile.defs (hunk 1 above)
Comment (timfoster) [reply Makefile.defs:72]: That's fair enough (I looked for that, but mustn't have looked hard enough!)

File: Makefile.defs (hunk 2 above)
Comment (jlevon) [reply Makefile.defs:66]: sure

**Pull request merged**